pkg/vex/vex.go

```diff
@@ -229,24 +244,58 @@ func SortDocuments(docs []*VEX) []*VEX {
 
 // Open tries to autodetect the vex format and open it
 func Open(path string) (*VEX, error) {
+	logrus.Info("Abriendo")
 	data, err := os.ReadFile(path)
 	if err != nil {
 		return nil, fmt.Errorf("opening VEX file: %w", err)
 	}
 
+	if bytes.Contains(data, []byte(Context+"/"+SpecVersion)) {
+		logrus.Info("opening current vex")
+		return Parse(data)
+	} else if bytes.Contains(data, []byte(Context)) {
```

Comment:
I think it's possible that this `bytes.Contains` check isn't as precise as we'd want, and as lib consumers would want. I'd suggest we try parsing in a data-structure-aware way to grab the context, and do an exact match on its value.

Reply:
I'd like to avoid parsing the whole document and instead do it this way to detect the OpenVEX version early to only open against the spec once. I can strengthen the regex if that is better. Also, if we can follow up on this one I would really appreciate it (to unlock the other PRs) :pray:

Reply:
Yeah we can make this a follow-up — but this one's important to address. `bytes.Contains` is logically problematic because it'll find the string anywhere in the data, including in other field values.

I'd be curious to learn more about your concern with parsing. In the existing code, it looks like you've already read the entire file into memory by the time you're doing the check.

On top of that, `bytes.Contains` will search through the byte slice (the entire file, here) until it finds the search string. And since you have two calls to `bytes.Contains` here, you'll end up search the whole file twice if the string can't be found. If you go the parsing route, you can parse just once, and then check the context's field's value as many times as you want afterwards.

One note is that it's possible to do a "partial parse", too. So you could do a preliminary parse using a struct that only has the field you want to examine. Then you can check its value how you want, and come back and re-parse with the full struct once you know what exact spec to which the data must now conform.

Last thought: As I think about this, I'm concerned about the security implications of doing this parsing in an imprecise manner. It shouldn't be possible for someone to put a context string into the body of a VEX statement and completely recolor how the VEX document is interpreted by a security tool. (I don't see this as a likely attack, but to me it's a hint that data isn't being handled properly).

Reply:
@puerco, assuming this isn't too much work, this does sound like it could be worth addressing here in this PR. If not, another PR ASAP should probably handle this.

I too am curious about the performance concern you bring up with a parsing. Perhaps a timing test or two could help inform this discussion? More generally, I'm not sure it's worth focusing on performance speed at this point in `go-vex`'s lifecycle. That can come later :)

The partial parse option sounds promising. 

And I do think that the security implication should be taken seriously, as least as a heuristic.

Thanks, @luhring, for the careful review!

Reply:
Thanks for pointing this out @luhring:
> I'm concerned about the security implications of doing this parsing in an imprecise manner

 I've opened https://github.com/openvex/go-vex/pull/50 reworking how the context locator is parsed :rocket: